src/Init/Prelude.lean

```diff
@@ -2262,8 +2262,13 @@ abbrev Unexpander := Syntax → UnexpandM Syntax
 
 -- unexpanders should not need to introduce new names
 instance : MonadQuotation UnexpandM where
-  getRef              := pure Syntax.missing
-  withRef             := fun _ => id
+  getRef              := get
+  withRef             := fun ref m => do
+    let orig ← get
+    set ref
+    let res ← m
+    set orig
+    return res
```

Comment:
```suggestion
    try m finally set orig
```

Reply:
This change would actually break compilation since we are working in Prelude.lean where this is not available yet (I'll revert it).